### DISCUSSION TITLE
Suggestions for Waveform Component of UI

### DISCUSSION THREAD

**Discussion by ThePerfectComputer** [emilk/egui#2045] [General]
I am trying to create the equivalent of [GTKWave](http://gtkwave.sourceforge.net) using eGUI. I'm wondering what type of UI elements I should be using to create the waveform part below. Note that waveforms can be several gigabytes large - so I've already written the code that allows you to quickly query the value of a a waveform at a certain point in time. The procedure I envision is that one would one would convert the pixels along the width of the waveform frame into timestamps, compute a value at every timestamp, and then draw that value. The user would also be able to scroll left and right which means that at every update, the visible portion of the waveform must be re-computed.

So basically, my question here is: are there any parts of egui that I should be looking at in particular to accomplish this?

![image](https://user-images.githubusercontent.com/103011002/190396966-6321d924-d2b0-4ffd-8542-49eb70184218.png)
